FirebaseInstallations/Source/Library/InstallationsIDController/FIRInstallationsIDController.m

```diff
@@ -114,4 +129,43 @@ - (instancetype)initWithGoogleAppID:(NSString *)appID
   return promise;
 }
 
+#pragma mark - FID registration
+
+- (FBLPromise<FIRInstallationsItem *> *)registerInstallationIfNeeded:
+    (FIRInstallationsItem *)installation {
+  switch (installation.registrationStatus) {
+    case FIRInstallationStatusRegistered:
+      // Already registered. Do nothing.
+      return [FBLPromise resolvedWith:installation];
+
+    case FIRInstallationStatusUnknown:
+    case FIRInstallationStatusUnregistered:
+    case FIRInstallationStatusRegistrationInProgress:
+      // Registration required. Proceed.
+      break;
```

Comment:
does it have to return sth?

Reply:
@chliangGoogle  No, we just proceed with the method execution in these cases. Now the switch can be replaced by just 
```if (installation.registrationStatus == FIRInstallationStatusRegistered) { return return [FBLPromise resolvedWith:installation]; }```.
But, from my experience such conditions on enums often lead to logical errors when a new value is added or an old one is modified. In this case the modified value implicitly falls under the condition `registrationStatus != FIRInstallationStatusRegistered` no matter what it is. A `switch` with no `default` clause is a reliable guard for such issues. I think, it is worth to make the check a bit verbose to make the code more robust and safe for modifications.